### ISSUE TITLE
DPI problem

### ISSUE THREAD

**Issue by stumts** [RPCS3/rpcs3#4463]
Hi, guys.

I have a notebook with a NVIDIA GTX 850m. I've tried to use RPCS3 on my TV and it has happened:
![captura de tela de 2018-04-19 17-11-34](https://user-images.githubusercontent.com/30453661/39016341-842f876e-43f6-11e8-9c93-c76983b83428.png)

On my internal display it doesn't happen:
![captura de tela de 2018-04-19 17-14-29](https://user-images.githubusercontent.com/30453661/39016462-e9571da0-43f6-11e8-808a-2fcf809619cf.png)

I fixed it using this Environment Variable "QT_SCREEN_SCALE_FACTORS" that I found here [http://doc.qt.io/qt-5/highdpi.html](url). A set it to "0.333333". Look my terminal in the picture bellow:
![captura de tela de 2018-04-19 17-13-21](https://user-images.githubusercontent.com/30453661/39016724-abffa0b6-43f7-11e8-8da7-eaede118d8cb.png)

**System Specs:**
- RPCS3 v0.0.5-6656
- HP Envy 15;
- Graphic Card: GTX 850m;
- Graphic Driver: NV Proprietary Driver (nvidia-396);
- SO:Ubuntu 18.04

Does anyone have any ideia what's causing this?

**Comment by flash-fire** [comment]
Ugh.  Qt and HiDPI.  It never stops haunting us.

**Comment by Megamouse** [comment]
just wondering. is that TV your second screen or configured as your main screen?
Or do you switch to it while having RPCS3 open?
If yes: what happens when you boot into your OS and RPCS3 while already having the TV set up as main screen?

**Comment by Megamouse** [comment]
have you tried setting multiple factors with:
`QT_SCREEN_SCALE_FACTORS [list]`
or using
`QT_SCALE_FACTOR`

**Comment by stumts** [comment]
QT_SCREEN_SCALE_FACTORS [list] fix my problem:

**My TV as my main screen with QT_SCREEN_SCALE_FACTORS="0.333333;1":**
![captura de tela de 2018-04-25 10-54-58](https://user-images.githubusercontent.com/30453661/39250912-65fe2436-4878-11e8-9398-194b0767806e.png)

![captura de tela de 2018-04-25 10-55-52](https://user-images.githubusercontent.com/30453661/39250921-6cfa5e30-4878-11e8-895f-c0b29465bb4a.png)

**My Internal Display as my main screen with QT_SCREEN_SCALE_FACTORS="1;0.333333":**
![captura de tela de 2018-04-25 10-56-53](https://user-images.githubusercontent.com/30453661/39250978-9209d516-4878-11e8-8088-fcd9d2738f36.png)

![captura de tela de 2018-04-25 10-57-10](https://user-images.githubusercontent.com/30453661/39250994-9ccdc69c-4878-11e8-8e70-53f333e01671.png)

**QT_SCALE_FACTOR didn't work 'cause it change point sized fonts also:**
QT_SCALE_FACTOR=1
![captura de tela de 2018-04-25 11-08-46](https://user-images.githubusercontent.com/30453661/39251263-392f910a-4879-11e8-987d-3f4f624abcc9.png)

QT_SCALE_FACTOR=0.333333333333
![captura de tela de 2018-04-25 11-08-09](https://user-images.githubusercontent.com/30453661/39251282-47a5ce52-4879-11e8-889a-cbaf6980aa04.png)

I just don't understand what's happening. Is it a problem with my linux, my TV EDID, rpcs3 QT5 implementation or QT5 itself? 'cause it doesn't happen on Windows.

**Comment by Megamouse** [comment]
in case you are compiling your own builds you could try changing line 85 in main.cpp to
![image](https://user-images.githubusercontent.com/23019877/39252255-9148885c-48a5-11e8-9edd-709a6ceed948.png)
and see what happens.
I don't remember why we don't use it on Linux. There is an old issue where we discussed this. (On Windows it is less accurate than the current implementation, although I heard rumours of this getting fixed in the next Qt update).

Any way, this is most likely a bug in Qt and/or the Linux distro

**Comment by stumts** [comment]
I've tried with QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling) and nothing changed:
![captura de tela de 2018-04-25 12-11-34](https://user-images.githubusercontent.com/30453661/39255163-41ee3162-4882-11e8-9b0f-a7056ee77f78.png)

Then I tried with QCoreApplication::setAttribute(Qt::AA_DisableHighDpiScaling) and it worked like a charm:
![captura de tela de 2018-04-25 12-12-08](https://user-images.githubusercontent.com/30453661/39255278-80df1f94-4882-11e8-83a0-3860b3c25ea7.png)

**Comment by Megamouse** [comment]
what if you just remove the whole section and don't set any scaling options.

**Comment by stumts** [comment]
I make some tests changing my screen scale: 
- I removed the whole section;
- I set qputenv("QT_AUTO_SCREEN_SCALE_FACTOR", "0");
- I set qputenv("QT_AUTO_SCREEN_SCALE_FACTOR", "1") + QCoreApplication::setAttribute(Qt::AA_DisableHighDpiScaling) 

The results are the same:
![captura de tela de 2018-04-25 12-56-38](https://user-images.githubusercontent.com/30453661/39258135-9737e9a4-4889-11e8-891e-e4faed548848.png)
![captura de tela de 2018-04-25 12-57-24](https://user-images.githubusercontent.com/30453661/39258145-9b5d8610-4889-11e8-98d4-62406e6c0b2b.png)

And I also tested with qputenv("QT_AUTO_SCREEN_SCALE_FACTOR", "1"):
![captura de tela de 2018-04-25 13-00-21](https://user-images.githubusercontent.com/30453661/39258242-e9016454-4889-11e8-9a5d-01c33d7d4fde.png)
![captura de tela de 2018-04-25 13-00-39](https://user-images.githubusercontent.com/30453661/39258262-f0a51200-4889-11e8-9c79-3bcb08123c79.png)

The conclusion that I got is the pixmaps aren't matching with my TV scale, but the fonts are.

**Comment by Megamouse** [comment]
you could try to move the `QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);` command into the win32 section (or just remove it).
Maybe it has some influence

**Comment by stumts** [comment]
Moving the QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps); command into the win32 section or removing it, the results are the same, big and ugly icons:
![captura de tela de 2018-04-25 13-36-11](https://user-images.githubusercontent.com/30453661/39259789-c4445b36-488d-11e8-970e-4bb6c701075b.png)

I think it's really a Qt5 bug.

**Comment by Megamouse** [comment]
yes

**Comment by stumts** [comment]
I think for now the best solution would be disable HiDPI with qputenv("QT_AUTO_SCREEN_SCALE_FACTOR", "0"); comand, but it's up to you. It's not working anyway or with weird behavior.

**Comment by Megamouse** [comment]
Don't forget, It might only be wrong on your distro

**Comment by stumts** [comment]
Yes, it's true. I'll make tests in others distros.

**Comment by stumts** [comment]
Hi guys

I tried Qt 5.11 beta but I can't build with it. It make compile fail. 

I have a clue what's is wrong. In Qt web page [http://doc.qt.io/qt-5/highdpi.html](url) it's said that the enviroment QT_SCREEN_SCALE_FACTORS it's used for monitors with wrong edid. In fact my monitor has wrong edid values, for exemple, the screen size is wrong. I've fixed my edid, fixed its checksun but I couldn'd apply it. I can't replace it directly and I can't apply it "on the fly" 'cause my pc has an nvdia with optmus tecnology, there for, my monitor it's not showed in nvidia-settings. With someone could test my bet, it would be amazing. 

Anyway, I fixed my problem adding QT_AUTO_SCREEN_SCALE_FACTOR=0 in my enviroment.
